src/entities/currency.ts

```diff
@@ -1,37 +1,59 @@
 import JSBI from 'jsbi'
-
+import { Token } from './token'
 import { SolidityType } from '../constants'
 import { validateSolidityTypeInstance } from '../utils'
 
 /**
- * A currency is any fungible financial instrument on Ethereum, including Ether and all ERC20 tokens.
- *
- * The only instance of the base class `Currency` is Ether.
+ * Represents the native currency of the chain on which it resides, e.g. ETH, AVAX
  */
-export class Currency {
+export class NativeCurrency {
   public readonly decimals: number
   public readonly symbol?: string
   public readonly name?: string
+  public readonly isNative: true = true
+  public readonly isToken: false = false
+  public readonly chainId: number
 
   /**
-   * The only instance of the base class `Currency`.
-   */
-  public static readonly CAVAX: Currency = new Currency(18, 'AVAX', 'Avalanche')
-
-  /**
-   * Constructs an instance of the base class `Currency`. The only instance of the base class `Currency` is `Currency.ETHER`.
+   * Constructs an instance of the base class `NativeCurrency`.
+   * @param chainId the chain ID on which this currency resides
    * @param decimals decimals of the currency
    * @param symbol symbol of the currency
    * @param name of the currency
    */
-  protected constructor(decimals: number, symbol?: string, name?: string) {
+  constructor(chainId: number, decimals: number, symbol?: string, name?: string) {
     validateSolidityTypeInstance(JSBI.BigInt(decimals), SolidityType.uint8)
-
+    this.chainId = chainId
     this.decimals = decimals
     this.symbol = symbol
     this.name = name
   }
+
+  public equals(other: NativeCurrency): boolean {
+    return other.isNative && other.chainId === this.chainId
+  }
 }
 
-const CAVAX = Currency.CAVAX
-export { CAVAX }
```

Comment:
The frontend currently relies on `CAVAX`

Reply:
Good suggestion on this!
The old `CAVAX` is type `Currency` while the new would be type `NativeCurrency` I hope that doesn't break anything.